beacon_node/beacon_chain/src/builder.rs

```diff
@@ -816,6 +816,24 @@ where
             ));
         }
 
+        // Check if the store is within the weak subjectivity period
+        if let Some(ws_checkpoint) = self.chain_config.weak_subjectivity_checkpoint {
+            let Ok(finalized_block) = fork_choice.get_finalized_block() else {
+                panic!("TODO(ws)")
+            };
+            if !store.is_within_weak_subjectivity_period(
```

Comment:
Just use the `head_snapshot.beacon_state` to compute the weak subjectivity period. You don't need to fetch the head state's finalize state. For WS sync, the head state = the weak subjectivity checkpoint state. You don't need to fetch any more data:
- compute ws_period from head state
- get current epoch from clock
- check `current_epoch <= head_state.epoch + ws_period`

No need to involve the store here